src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -0,0 +1,640 @@
+        runResultsList.Add(results.MakeSummary(gameProperties.GameWorld.Map, true));
+        speciesHistoryList.Add(
+            gameProperties.GameWorld.Species.ToDictionary(pair => pair.Key, pair => (Species)pair.Value.Clone()));
+
+        // Add check box to history container
+        historyListMenu.AddItem(currentGeneration.ToString(), false, Colors.White);
+        historyListMenu.CreateElements();
+
+        currentGenerationLabel.Text = currentGeneration.ToString();
+    }
+
+    /// <summary>
+    ///   Run one step sync
+    /// </summary>
+    private void OnRunOneStepButtonPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        if (autoEvoRun?.Aborted != false || autoEvoRun.Finished)
+        {
+            autoEvoRun = new AutoEvoRun(gameProperties.GameWorld, autoEvoConfiguration);
+        }
+
+        // To avoid concurrent steps
+        autoEvoRun.FullSpeed = false;
+        autoEvoRun.OneStep();
+        abortButton.Disabled = false;
+    }
+
+    /// <summary>
+    ///   Abort the current run
+    /// </summary>
+    private void OnAbortButtonPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        if (autoEvoRun?.WasSuccessful == false)
+            autoEvoRun.Abort();
+
+        finishOneGenerationButton.Disabled = false;
+        runOneStepButton.Disabled = false;
+    }
+
+    private void HistoryListMenuIndexChanged(int index)
+    {
+        if (generationDisplayed == index && speciesListMenu.Popup.GetItemCount() > 0)
+            return;
+
+        historyListMenu.Text = index.ToString();
+
+        generationDisplayed = index;
+        UpdateAutoEvoReport();
+        UpdateSpeciesList();
+    }
+
+    private void UpdateAutoEvoReport()
+    {
+        if (generationDisplayed < runResultsList.Count && generationDisplayed > -1)
+        {
+            autoEvoResultsLabel.ExtendedBbcode = runResultsList[generationDisplayed].ToString();
+        }
+    }
+
+    private void UpdateSpeciesList()
+    {
+        speciesListMenu.ClearAllItems();
+
+        foreach (var pair in speciesHistoryList[generationDisplayed].OrderBy(p => p.Value.FormattedName))
+        {
+            speciesListMenu.AddItem(pair.Value.FormattedName, false, Colors.White);
+        }
+
+        speciesListMenu.CreateElements();
+    }
+
+    private void SpeciesListMenuIndexChanged(int index)
+    {
+        var speciesName = speciesListMenu.Popup.GetItemText(index);
+        var species = speciesHistoryList[generationDisplayed].Values.First(p => p.FormattedName == speciesName);
+
+        if (species == speciesPreview.PreviewSpecies)
+            return;
+
+        UpdateSpeciesPreview(species);
+    }
+
+    private void UpdateSpeciesPreview(Species species)
+    {
+        speciesListMenu.Text = species.FormattedName;
+        speciesPreview.PreviewSpecies = species;
+        hexPreview.PreviewSpecies = species as MicrobeSpecies;
```

Comment:
I'd add an if here for the cast and print an error if the species is an unknown type to preview.